common/src/main/java/com/example/android/uamp/media/library/AlbumArtContentProvider.kt

```diff
@@ -53,28 +49,22 @@ class AlbumArtContentProvider : ContentProvider() {
         val context = this.context ?: return null
         val remoteUri = uriMap[uri] ?: throw FileNotFoundException(uri.path)
 
-        val file = File(context.cacheDir, uri.path)
-        return if (!file.exists()) {
-            GlobalScope.launch(Dispatchers.IO) {
-                // Use Glide to download the album art
-                val cacheFile = Glide.with(context)
-                    .asFile()
-                    .load(remoteUri)
-                    .submit()
-                    .get()
-
-                // Rename the file Glide created to match our own scheme.
-                cacheFile.renameTo(file)
-
-                // Notify the caller that the artwork has been updated.
-                context.contentResolver.notifyChange(uri, null)
-            }
-
-            // Provide placeholder art until the correct art can be downloaded.
-            context.assets.openFd("default_art.png").parcelFileDescriptor
-        } else {
-            ParcelFileDescriptor.open(file, ParcelFileDescriptor.MODE_READ_ONLY)
+        var file = File(context.cacheDir, uri.path)
+
+        if (!file.exists()) {
+            // Use Glide to download the album art.
+            val cacheFile = Glide.with(context)
+                .asFile()
+                .load(remoteUri)
+                .submit()
+                .get()
```

Comment:
Maybe add a timeout here (so `.get(TIMEOUT, TimeUnit.MILLISECONDS)` with some 'reasonable' value for that?

Alternatively, add a comment that one _should_ do this in a production app, but it seems "simple enough"?

Though it might then make sense to actually extract that to a function that's like `download(uri)` so it can encapsulate the try/catch for the timeout and return either the file or null.

WDYT?

Reply:
Timeout = great idea, added. 

try/catch, return null = not so keen. If there's a `TimeoutException` as the caller I'd want to catch it so I know that there's a problem with the server, rather than just that the file doesn't exist.